src/v/cloud_storage/remote_path_provider.cc

```diff
@@ -32,30 +32,37 @@ remote_path_provider remote_path_provider::copy() const {
 
 ss::sstring remote_path_provider::topic_manifest_prefix(
   const model::topic_namespace& topic) const {
+    const auto& tp_ns = _topic_namespace_override.value_or(topic);
     if (label_.has_value()) {
-        return labeled_topic_manifest_prefix(*label_, topic);
+        return labeled_topic_manifest_prefix(*label_, tp_ns);
     }
-    return prefixed_topic_manifest_prefix(topic);
+    return prefixed_topic_manifest_prefix(tp_ns);
 }
 
 ss::sstring remote_path_provider::topic_manifest_path(
   const model::topic_namespace& topic, model::initial_revision_id rev) const {
+    const auto& tp_ns = _topic_namespace_override.value_or(topic);
     if (label_.has_value()) {
-        return labeled_topic_manifest_path(*label_, topic, rev);
+        return labeled_topic_manifest_path(*label_, tp_ns, rev);
     }
-    return prefixed_topic_manifest_bin_path(topic);
+    return prefixed_topic_manifest_bin_path(tp_ns);
 }
 
 std::optional<ss::sstring> remote_path_provider::topic_manifest_path_json(
   const model::topic_namespace& topic) const {
     if (label_.has_value()) {
         return std::nullopt;
     }
-    return prefixed_topic_manifest_json_path(topic);
+    const auto& tp_ns = _topic_namespace_override.value_or(topic);
+    return prefixed_topic_manifest_json_path(tp_ns);
 }
 
 ss::sstring remote_path_provider::partition_manifest_prefix(
-  const model::ntp& ntp, model::initial_revision_id rev) const {
+  model::ntp ntp, model::initial_revision_id rev) const {
```

Comment:
I doubt it's worth optimizing, but it's a shame this can't avoid copies in the common path. Ultimately it probably doesn't matter anyway since this is generating a brand new string

Reply:
Yeah, I was looking for a way to avoid copies, but couldn't really find the best way to do so here, with the exception of adding some sort of `ntp_view` type, or changing around the call sites to use `topic_namespace_view/partition` as arguments while adding some new `path()` functions that accept those types as well.

Future work, perhaps.